File: core-bc/modules/client/src/app/model/contentField.ts
export class ContentField {
  name: string;
  label: string;
  contentFieldValue: { data?: string, image?: string };
  dataType: 'integer' | 'string' | 'image';
  nestedContentFields: ContentField[];

  public getHeading(): string {
    return this.contentFieldValue.data;
  }


}

File: core-bc/composites/types/src/main/java/se/vgregion/vatskenutrition/model/v2/ContentField.java
package se.vgregion.vatskenutrition.model.v2;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ContentField {

    private ContentFieldValue contentFieldValue;
    private String dataType;
    private String label;
    private String name;
    private List<ContentField> nestedContentFields;
}
